components/ActivityInfo.js

```diff
@@ -1,46 +1,222 @@
 
 export default function ActivityInfo({ activity }) {
+  const [isEditing, setIsEditing] = useState(false);
+  const [formTitle, setFormTitle] = useState(activity.title);
+  const [formCategories, setFormCategories] = useState(
+    activity.categories.map((c) => c._id)
+  );
+  const [formArea, setFormArea] = useState(activity.area || "");
+  const [formCountry, setFormCountry] = useState(activity.country || "");
+  const [formLat, setFormLat] = useState(activity.coordinates?.lat ?? "");
+  const [formLng, setFormLng] = useState(activity.coordinates?.lng ?? "");
+  const [error, setError] = useState("");
+
+  const { data: allCategories } = useSWR("/api/categories", fetcher);
+  const { mutate } = useSWRConfig();
+
+  if (!isEditing) {
+    return (
+      <>
+        <h1>See more Details of your selected activity</h1>
+
+        <BackButton />
+        <h2>{activity.title}</h2>
+
+        <StyledImage
+          alt={activity.title}
+          src={activity.imageUrl}
+          width={1200}
+          height={900}
+        />
+
+        <StyledBlock>
+          <h3>Description:</h3>
+          <p>{activity.description}</p>
+        </StyledBlock>
+
+        <StyledBlock>
+          <h3>Categories:</h3>
+          <ul>
+            {activity.categories.map((category) => (
+              <li key={category._id}>{category.name}</li>
+            ))}
+          </ul>
+        </StyledBlock>
+
+        <StyledBlock>
+          <h3>Location:</h3>
+
+          <p>Area: {activity.area || "—"}</p>
+          <p>Country: {activity.country || "—"}</p>
+
+          {activity.coordinates?.lat != null &&
+            activity.coordinates?.lng != null && (
+              <>
+                <p>Latitude: {activity.coordinates.lat}</p>
+                <p>Longitude: {activity.coordinates.lng}</p>
+
+                <ActivityMap
+                  lat={activity.coordinates.lat}
+                  lng={activity.coordinates.lng}
+                />
+              </>
+            )}
+        </StyledBlock>
+
+        <button onClick={() => setIsEditing(true)}>Edit</button>
+      </>
+    );
+  }
+
   return (
     <>
-      <h1>See more Details of your selected activity</h1>
-
-      <BackButton />
-      <h2>{activity.title}</h2>
-
-      <StyledImage
-        alt={activity.title}
-        src={activity.imageUrl}
-        width={1200}
-        height={900}
-      />
-
-      <StyledP>
-        <h3>Description:</h3>
-        <p>{activity.description}</p>
-      </StyledP>
-
-      <StyledP>
-        <h3>Categories:</h3>
-        <ul>
-          {activity.categories.map((category) => (
-            <li key={category._id}>{category.name}</li>
-          ))}
-        </ul>
-      </StyledP>
-
-      <StyledP>
-        <h3>Location:</h3>
-        <p>📍 Area: {activity.area}</p>
-        <p>🌍 Country: {activity.country}</p>
-      </StyledP>
+      <h2>Edit Activity</h2>
+
+      {error && <p style={{ color: "red" }}>{error}</p>}
+
+      <form
```

Comment:
Please do not render the edit form as a part of the ActivityInfo. That is a bad structure. The better concept is called `Separation of concerns`. 
The purpose of a component named 'ActivityInfo' is to render infos about an activity.
The purpose of a component named 'ActivityForm' is ... ? Yes, to render an form.
So why don't you reuse the 'ActivityForm' component that you already have? That would be very beneficial for code structure, code maintenance, code readability, ...